### PR TITLE
Updates in the examples, class BilinearForm and related [okina-bilinearform-updates]

### PR DESCRIPTION
**Pull request by v-dobrev**
Updates in the BilinearForm and related classes. Small updates in ex1, ex1p, ex6, ex6p.

There are still some issues:
* ex6 sometimes fails with 'cuda' (fixed with the mm erase)
* ex6p fails with PA even with 'cpu' (fixed)

### CODE REVIEW

File: examples/ex1.cpp [hunk 1]
```diff
@@ -73,8 +73,8 @@ int main(int argc, char *argv[])
                   " isoparametric space.");
    args.AddOption(&static_cond, "-sc", "--static-condensation", "-no-sc",
                   "--no-static-condensation", "Enable static condensation.");
-   args.AddOption(&pa, "-p", "--pa", "-no-p", "--no-pa",
-                  "Enable Partial Assembly.");
+   args.AddOption(&pa, "-pa", "--partial-assembly", "-no-pa",
+                  "--no-partial-assembly", "Enable Partial Assembly.");
    args.AddOption(&device, "-d", "--device",
```
Comment (tzanio) [inline examples/ex1.cpp]: 👍

File: examples/ex1.cpp [hunk 2]
```diff
@@ -176,31 +175,30 @@ int main(int argc, char *argv[])
    a->Assemble();
 
    Vector B, X;
-   Operator *A;
-   if (!pa) { A = new SparseMatrix; }
+   OperatorHandle Ah;
+   a->FormLinearSystem(ess_tdof_list, x, *b, Ah, X, B);
+   Operator &A = *Ah.Ptr();
```
Comment (tzanio) [inline examples/ex1.cpp]: I like that this eliminates the pointer, but I still think this will be confusing for a lot of users...

Comment (v-dobrev) [reply examples/ex1.cpp]: We can add a method `Ah.Oper()` or `Ah.Get()` that returns `Operator &` if `Ah.Ptr()` is confusing.

Comment (tzanio) [reply examples/ex1.cpp]: I meant more: "why do we need two `Operator` objects and what is the difference between them?".

File: fem/bilinearform_ext.hpp [hunk 3]
```diff
@@ -39,99 +39,108 @@ class ElemRestriction: public Operator
    void MultTranspose(const Vector &x, Vector &y) const;
 };
 
+
+class BilinearFormExtension : public Operator
```
Comment (tzanio) [inline fem/bilinearform_ext.hpp]: Having this class is definitely cleaner.

File: fem/bilinearform_ext.hpp [hunk 4]
```diff
@@ -39,99 +39,108 @@ class ElemRestriction: public Operator
...
 public:
-   MFBilinearFormExtension(BilinearForm *form);
+   MFBilinearFormExtension(BilinearForm *form)
+      : BilinearFormExtension(form) { }
 
    /// TODO
-   void AddDomainIntegrator(BilinearFormIntegrator*) {}
    void Assemble() {}
-   void FormSystemOperator(const Array<int> &ess_tdof_list, Operator *&A) {}
+   void FormSystemMatrix(const Array<int> &ess_tdof_list, OperatorHandle &A) {}
```
Comment (tzanio) [inline fem/bilinearform_ext.hpp]: `FormSystemMatrix` is quite a misnomer in this case, which is why we thought `FormSystemOperator` may be better... (I don't insist on changing back the name though, I can live with `FormSystemMatrix`)

**Comment by v-dobrev** [comment]
I think I fixed the issue with `ex6p` in https://github.com/mfem/mfem/pull/851/commits/14f063d60818822345e9832105323a40a38eb1dd.

**Comment by camierjs** [comment]
Yes, nice, I like it better!

**Pull request merged**